### PR TITLE
Remove isAndroid check from setupExtractIncludeProtosTask

### PR DESCRIPTION
**Pull request by ejona86**
Have Java and Android pass in the appropriate objects instead of looking them up. This was the last isAndroid check in ProtobufPlugin excepting apply() itself.

### CODE REVIEW

File: src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy [hunk 1]
```diff
@@ -290,15 +294,23 @@ class ProtobufPlugin implements Plugin<Project> {
...
-            variant.testedVariant.compileConfiguration.incoming.artifactView {
+      FileCollection testClassPathConfig = project.objects.fileCollection()
+      if (Utils.isTest(variant.name)) {
+        // TODO(zhangkun83): Android sourceSet doesn't have compileClasspath. If it did, we
+        // haven't figured out a way to put source protos in 'resources'. For now we use an
+        // ad-hoc solution that manually includes the source protos of 'main' and its
+        // dependencies.
+        testClassPathConfig = project.android.sourceSets['main'].proto.sourceDirectories
+        if (variant.hasProperty("testedVariant")) {
+          testClassPathConfig += variant.testedVariant.compileConfiguration.incoming.artifactView {
```
Comment (rougsig) [inline src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy:305]: In common gradle works with mutable objects. I think it's not expected to add a testedVariant artifact to the main sourceSet. I will check, testClassPathConfig is mutable or not, tomorrow.

Comment (ejona86) [reply src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy:305]: Although it is a FileCollection which should be immutable (from our API perspective; we'd need a ConfigurableFileCollection or the like to change it), if += can be overloaded that could modify the original object in general be dangerous. I can repeat the variable name, although it is unweildy. But from https://groovy-lang.org/operators.html#Operator-Overloading , it looks like += can't be overridden (unlike C++ and .Net), so maybe this is safe? (Dunno if Kotlin has overloading.)

Comment (rougsig) [reply src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy:305]: All works as should works, Tested artifact is not added to main sourceSet. I see in logs different instances.

From sources:
```groovy
public FileCollection plus(FileCollection collection) {
  return new UnionFileCollection(new FileCollectionInternal[]{this, (FileCollectionInternal)collection});
}
```

Maybe replacing `+=` to `from` can improve readability:
```groovy
ConfigurableFileCollection testClassPathConfig = project.objects.fileCollection()
if (Utils.isTest(variant.name)) {
  testClassPathConfig.from(project.android.sourceSets['main'].proto.sourceDirectories) // <-- replace += to .from
  if (variant.hasProperty("testedVariant")) {
    testClassPathConfig.from(variant.testedVariant.compileConfiguration.incoming.artifactView {  // <-- replace += to .from
      attributes {
        it.attribute(artifactType, "jar")
      }
    }.files)
  }
}
```

**Review by rougsig — APPROVED** [review]

File: src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy (hunk 1 above)
Comment (ejona86) [reply src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy:305]: I see nothing to suggest += can be overridden to mutate in Groovy (as the overloading is the same as `foo = bar + baz`). So += is simply + followed by =. `=` and `++`/`--` seem to be the potentially-mutating operators. Thus, I think this is idiomatic; the += in Gradle behaves the same way and creates a new instance.

**Pull request merged**